crates/aptos-crypto/src/ed25519/mod.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides an API for the PureEdDSA signature scheme over the Ed25519 twisted
+//! Edwards curve as defined in [RFC8032](https://tools.ietf.org/html/rfc8032).
+//!
+//! Signature verification also checks and rejects non-canonical signatures.
+//!
+//! # Examples
+//!
+//! ```
+//! use aptos_crypto_derive::{CryptoHasher, BCSCryptoHash};
+//! use aptos_crypto::{
+//!     ed25519::*,
+//!     traits::{Signature, SigningKey, Uniform},
+//!     test_utils::KeyPair
+//! };
+//! use rand::{rngs::StdRng, SeedableRng};
+//! use rand_core::OsRng;
+//! use serde::{Serialize, Deserialize};
+//!
+//! #[derive(Serialize, Deserialize, CryptoHasher, BCSCryptoHash)]
+//! pub struct TestCryptoDocTest(String);
+//! let message = TestCryptoDocTest("Test message".to_string());
+//!
+//! let mut rng = OsRng;
+//! let kp = KeyPair::<Ed25519PrivateKey, Ed25519PublicKey>::generate(&mut rng);
+//!
+//! let signature = kp.private_key.sign(&message);
+//! assert!(signature.verify(&message, &kp.public_key).is_ok());
+//! ```
+
+/// The length of the Ed25519PrivateKey
+pub const ED25519_PRIVATE_KEY_LENGTH: usize = ed25519_dalek::SECRET_KEY_LENGTH;
+/// The length of the Ed25519PublicKey
+pub const ED25519_PUBLIC_KEY_LENGTH: usize = ed25519_dalek::PUBLIC_KEY_LENGTH;
+/// The length of the Ed25519Signature
+pub const ED25519_SIGNATURE_LENGTH: usize = ed25519_dalek::SIGNATURE_LENGTH;
+
+/// The order of ed25519 as defined in [RFC8032](https://tools.ietf.org/html/rfc8032).
+const L: [u8; 32] = [
+    0xed, 0xd3, 0xf5, 0x5c, 0x1a, 0x63, 0x12, 0x58, 0xd6, 0x9c, 0xf7, 0xa2, 0xde, 0xf9, 0xde, 0x14,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x10,
+];
+
+pub mod ed25519_keys;
+pub mod ed25519_sigs;
+
+#[cfg(any(test, feature = "fuzzing"))]
+pub use ed25519_keys::keypair_strategy;
+
+pub use ed25519_keys::{Ed25519PrivateKey, Ed25519PublicKey};
+pub use ed25519_sigs::Ed25519Signature;
```

Comment:
can we actually improt these also as 
ed25519::private_key, ed25519::public_key, and ed25519::signature

which will be consistent with bls and x25519

also not sure why we split the files up 🤷‍♂️

Reply:
**1.** The reason this PR is just splitting files is I was adding extra proof-of-knowledge (PoK) functionality (which I removed yesterday due to unclear use-case) and I wanted to structure this similar to the `bls12381/` directory. Also, the single `ed25519.rs` file was rather big.

**2.** Sure, I assume you mean capitalized?

In BLS, we have:

```
pub use bls12381_keys::{PrivateKey, PublicKey};
pub use bls12381_pop::ProofOfPossession;
pub use bls12381_sigs::Signature;
```

In x25519, we also export `x25519::PublicKey` and `x25519::PrivateKey`.

Reply:
Woops capitalized indeed! Thanks for the explanation!